internals/approval_defs.py

```diff
@@ -116,3 +117,35 @@ def fields_approvable_by(user):
 def is_valid_field_id(field_id):
   """Return true if field_id is a known field."""
   return field_id in APPROVAL_FIELDS_BY_ID
+
+
+def is_approved(approval_values, field_id):
+  """Return true if we have all needed APPROVED values and no NOT_APPROVED."""
+  count = 0
+  for av in approval_values:
+    if av.state == models.Approval.APPROVED:
+      count += 1
+    elif av.state == models.Approval.NOT_APPROVED:
+      return False
+  afd = APPROVAL_FIELDS_BY_ID[field_id]
+
+  if afd.rule == ONE_LGTM:
+    return count >= 1
+  elif afd.rule == THREE_LGTM:
+    return count >= 3
+  else:
+    logging.error('Unexpected approval rule')
+    return False
+
+
+def is_resolved(approval_values, field_id):
+  """Return true if the review is done (approved or not approved)."""
```

Comment:
very optional nit:
```suggestion
  # Any NOT_APPROVED value means that the review is no longer pending.
  for av in approval_values:
    if av.state == models.Approval.NOT_APPROVED:
      return True
  return is_approved(approval_values, field_id)
```

Reply:
and perhaps remove line 128 and 129 if above is accepted

Reply:
Thanks for the comment.  I thought about it and I think that the original doc string is more accurate because an APPROVED value is also no longer pending.